### ISSUE TITLE
Expense - Approve button appears in archived expense report

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#38229]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.51-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- Admin and employee are in the Collect workspace.
- The Approval mode in the Collect workspace is Submit and Close.
1. Go to staging.new.expensify.com.
2. [Employee] Create a manual request in the workspace chat.
3. [Employee] Submit the report,
4. [Admin] Go to workspace chat with employee.
5. [Admin] Click Approve button.
6. [Admin] Click on the header subtitle to return to main chat.
7. {Admin] Click on expense preview to go to expense report.

## Expected Result:
In Step 5, Approve button should not appear in expense report since the expense report is archived.

## Actual Result:
In Step 5, Approve button appears in expense report when the expense report is archived.
When clicking Approve button, the expense report is no longer archived and back to archived state when reopening expense report (Step 7).

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/24837b59-0059-46a7-9923-e9fe90638d7d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
@slafortune I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by slafortune** [comment]
Submit and close should not/doesn't need to show the `Approve` button - Adding to #wavecollect

**Comment by rojiphil** [comment]
> Could you expand on why your changes to `submitReport` would be necessary here?

@Ollyws I agree that the report gets archived immediately by BE when the report is submitted in `Submit and Close` approval mode. But that's when it is online. When offline, changes to `submitReport` would be necessary to ensure that the app behaves as expected in `Submit and Close` approval mode. Hope this answer your query.

**Comment by rojiphil** [comment]
> But the report is archived agnostically, I can't see a situation in which the changes to submitReport would be necessary but maybe I'm missing something?

@Ollyws Sorry. I could have done better in communication here. I think changes to `submitReport` become necessary when the `submitter` and `approver` are the same user. Here is a test video for the same. Let me know if this is still not clear.

https://github.com/Expensify/App/assets/3004659/34c467c7-47ec-4020-8bff-c829969a4e13

**Comment by robertjchen** [comment]
Thanks for the discussion, and I also agree with the proposal 👍

**Comment by rojiphil** [comment]
@Ollyws [PR](https://github.com/Expensify/App/pull/39071) is ready for review.
